File: Csharp/CodeWars/src/Resolutions/ConvertStringToCamelCase.cs
using System.Globalization;

namespace CodeWars.Resolutions;

public class ConvertStringToCamelCase
{
    public static string ToCamelCase(string str) =>
        string.Concat(str
            .Replace("-", " ")
            .Replace("_", " ")
            .Split(" ")
            .Select((x, i) => i == 0 ? x : CultureInfo.CurrentCulture.TextInfo.ToTitleCase(x))
        );
}

File: TypeScript/Codewars/src/resolutions/Convert-string-to-camel-case.ts
export function toCamelCase(str: string): string {
	const regex = /[-_]/g;

	const newStr = str
		.replace(regex, " ")
		.split(" ")
		.map((value, index) => {
			if (index !== 0) {
				const firstLetter = (value[0] ?? "").toUpperCase();
				return firstLetter + value.slice(1);
			}

			return value;
		});

	return newStr.join("");
}
